scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py

```diff
@@ -0,0 +1,258 @@
+        for plot_info in self.plotted_workspace_information:
+            if plot_info.workspace_name == workspace_name:
+                axis = self.fig.axes[plot_info.axis]
+                workspace_name = plot_info.workspace_name
+                artist_info = axis.tracked_workspaces[workspace_name]
+                for ws_artist in artist_info:
+                    for artist in ws_artist._artists:
+                        if isinstance(artist, ErrorbarContainer):
+                            color = artist[0].get_color()
+                        else:
+                            color = artist.get_color()
+                        plot_kwargs = self._get_plot_kwargs(plot_info)
+                        plot_kwargs["color"] = color
+                        axis.replot_artist(artist, with_errors, **plot_kwargs)
+        self.redraw_figure()
+
+    def set_axis_xlimits(self, axis_number, xlims):
+        ax = self.fig.axes[axis_number]
+        ax.set_xlim(xlims[0], xlims[1])
+
+    def set_axis_ylimits(self, axis_number, ylims):
+        ax = self.fig.axes[axis_number]
+        ax.set_ylim(ylims[0], ylims[1])
+
+    def set_axes_limits(self, xlim, ylim):
+        plt.setp(self.fig.axes, xlim=xlim, ylim=ylim)
+
+    def autoscale_y_axes(self):
+        ymin = 1e9
+        ymax = -1e9
+        for axis in self.fig.axes:
+            ymin_i, ymax_i = self._get_y_axis_autoscale_limts(axis)
+            if ymin_i < ymin:
+                ymin = ymin_i
+            if ymax_i > ymax:
+                ymax = ymax_i
+
+        plt.setp(self.fig.axes, ylim=[ymin, ymax])
+
+    def autoscale_selected_y_axis(self, axis_number):
+        if axis_number >= len(self.fig.axes):
+            return
+        axis = self.fig.axes[axis_number]
+        bottom, top, = self._get_y_axis_autoscale_limts(axis)
+        axis.set_ylim(bottom, top)
+
+    def set_title(self, axis_number, title):
+        if axis_number >= self.number_of_axes:
+            return
+        axis = self.fig.axes[axis_number]
+        axis.set_title(title)
+
+    def get_axis_limits(self, axis_number):
+        xmin, xmax = self.fig.axes[axis_number].get_xlim()
+        ymin, ymax = self.fig.axes[axis_number].get_ylim()
+
+        return xmin, xmax, ymin, ymax
+
+    def redraw_figure(self):
+        self.fig.canvas.toolbar.update()
+        self._redraw_legend()
+        self.fig.tight_layout()
+        self.fig.canvas.draw()
+
+    def _redraw_legend(self):
+        for ax in self.fig.axes:
+            if ax.get_legend_handles_labels()[0]:
+                legend = ax.legend(prop=dict(size=5))
+                legend_set_draggable(legend, True)
+
+    def _get_plot_kwargs(self, workspace_info: WorkspacePlotInformation):
+        label = workspace_info.label
+        plot_kwargs = {'distribution': True, 'autoscale_on_update': False, 'label': label}
+        return plot_kwargs
+
+    @staticmethod
+    def _get_y_axis_autoscale_limts(axis):
+        bottom = 1e9
```

Comment:
why is top>bottom?

Reply:
This is just the initial guess of the axis limits, having the top as -1e9 means it will be replaced by the first calculated top limit (top_i).